src/main/java/com/minecolonies/compatibility/tinkers/TinkersWeaponHelper.java

```diff
@@ -0,0 +1,36 @@
+package com.minecolonies.compatibility.tinkers;
+
+import net.minecraft.item.ItemStack;
+import slimeknights.tconstruct.library.tools.SwordCore;
+import slimeknights.tconstruct.library.utils.ToolHelper;
+
+/**
+ * Class to check if certain tinkers items serve as weapons for the guards.
+ */
+public final class TinkersWeaponHelper
```

Comment:
Exactly.
There are even better implementations and i am working on those, but @Optional is the minimal requirement